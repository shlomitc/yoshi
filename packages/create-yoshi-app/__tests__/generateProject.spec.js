const path = require('path');
const tempy = require('tempy');
const { generateProject } = require('../src');
const Answers = require('../src/Answers');
const getFilesInDir = require('../src/getFilesInDir');

test('verify generation works as expected', () => {
  const tempDir = tempy.directory();
  const answers = new Answers({
    projectName: `test-project`,
    authorName: 'rany',
    authorEmail: 'rany@wix.com',
    organization: 'wix',
  });

  // override the templatePath to our fake template directory
  Object.defineProperty(answers, 'templatePath', {
    get: function() {
      return path.join(__dirname, './__fixtures__/fake-template/');
    },
  });

  generateProject(answers, tempDir);

  const files = getFilesInDir(tempDir);

  expect(files).toMatchSnapshot();
});
