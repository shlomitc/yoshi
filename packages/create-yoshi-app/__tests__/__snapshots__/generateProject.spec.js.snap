// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`verify generation works as expected 1`] = `
Object {
  ".gitignore": "",
  "TEST_PROJECT.json": "",
  "no-template.js": "console.log('foo');
",
  "package.json": "",
  "template.txt": "test-project
rany
rany@wix.com
wix

TEST_PROJECT
RANY
RANY_WIX_COM
WIX
",
}
`;
